timely_beliefs/beliefs/classes.py

```diff
@@ -226,15 +226,23 @@ def add_to_session(
         cls,
         session: Session,
         beliefs_data_frame: "BeliefsDataFrame",
+        expunge_session: bool = True,
+        allow_overwrite: bool = False,
         commit_transaction: bool = False,
     ):
         """Add a BeliefsDataFrame as timed beliefs to a database session.
 
-        :param session: the database session to use
-        :param beliefs_data_frame: the BeliefsDataFrame to be persisted
-        :param commit_transaction: if True, the session is committed
-                                   if False, you can still add other data to the session
-                                   and commit it all within an atomic transaction
+        If you are adding lots of beliefs, it's most efficient to use expunge_session=True and allow_overwrite=False
+
+        :param session:             the database session to use
+        :param beliefs_data_frame:  the BeliefsDataFrame to be persisted
+        :param expunge_session:     if True, all instances are removed from the session before adding beliefs
```

Comment:
1. We need to explain a bit better in which situations this can be useful, maybe after a discussion.
2. Why does one need to flush, and does that need to happen before or after?
3. This also leads to using `bulk_save_objects` which has a few consequences that matter, so it's worth mentioning. And why do we make that distinction?